beacon-chain/sync/data_columns_sampling.go

```diff
@@ -1,624 +0,0 @@
-	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/feed"
-	statefeed "github.com/prysmaticlabs/prysm/v5/beacon-chain/core/feed/state"
-	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/peerdas"
-	"github.com/prysmaticlabs/prysm/v5/beacon-chain/p2p"
-	"github.com/prysmaticlabs/prysm/v5/beacon-chain/p2p/types"
-	"github.com/prysmaticlabs/prysm/v5/beacon-chain/startup"
-	"github.com/prysmaticlabs/prysm/v5/config/params"
-	"github.com/prysmaticlabs/prysm/v5/consensus-types/blocks"
-	"github.com/prysmaticlabs/prysm/v5/crypto/rand"
-	eth "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
-	"github.com/prysmaticlabs/prysm/v5/runtime/version"
-)
-
-const PeerRefreshInterval = 1 * time.Minute
-
-type roundSummary struct {
-	RequestedColumns []uint64
-	MissingColumns   map[uint64]bool
-}
-
-// DataColumnSampler defines the interface for sampling data columns from peers for requested block root and samples count.
-type DataColumnSampler interface {
-	// Run starts the data column sampling service.
-	Run(ctx context.Context)
-}
-
-var _ DataColumnSampler = (*dataColumnSampler1D)(nil)
-
-// dataColumnSampler1D implements the DataColumnSampler interface for PeerDAS 1D.
-type dataColumnSampler1D struct {
-	sync.RWMutex
-
-	p2p           p2p.P2P
-	clock         *startup.Clock
-	ctxMap        ContextByteVersions
-	stateNotifier statefeed.Notifier
-
-	// nonCustodyGroups is a set of groups that are not custodied by the node.
-	nonCustodyGroups map[uint64]bool
-
-	// groupsByPeer maps a peer to the groups it is responsible for custody.
-	groupsByPeer map[peer.ID]map[uint64]bool
-
-	// peersByCustodyGroup maps a group to the peer responsible for custody.
-	peersByCustodyGroup map[uint64]map[peer.ID]bool
-
-	// columnVerifier verifies a column according to the specified requirements.
-	columnVerifier verification.NewDataColumnsVerifier
-
-	// custodyInfo contains the custody information of the node.
-	custodyInfo *peerdas.CustodyInfo
-}
-
-// newDataColumnSampler1D creates a new 1D data column sampler.
-func newDataColumnSampler1D(
-	p2p p2p.P2P,
-	clock *startup.Clock,
-	ctxMap ContextByteVersions,
-	stateNotifier statefeed.Notifier,
-	colVerifier verification.NewDataColumnsVerifier,
-	custodyInfo *peerdas.CustodyInfo,
-) *dataColumnSampler1D {
-	numberOfCustodyGroups := params.BeaconConfig().NumberOfCustodyGroups
-	peersByCustodyGroup := make(map[uint64]map[peer.ID]bool, numberOfCustodyGroups)
-
-	for i := range numberOfCustodyGroups {
-		peersByCustodyGroup[i] = make(map[peer.ID]bool)
-	}
-
-	return &dataColumnSampler1D{
-		p2p:                 p2p,
-		clock:               clock,
-		ctxMap:              ctxMap,
-		stateNotifier:       stateNotifier,
-		groupsByPeer:        make(map[peer.ID]map[uint64]bool),
-		peersByCustodyGroup: peersByCustodyGroup,
-		columnVerifier:      colVerifier,
-		custodyInfo:         custodyInfo,
-	}
-}
-
-// Run implements DataColumnSampler.
-func (d *dataColumnSampler1D) Run(ctx context.Context) {
```

Comment:
I know this is not technically part of the fulu spec, but why are we removing this ? Any harm in just leaving it in, in the event we end up having user run with sampling ( lower custody count) for bandwidth savings.

We spent a lot of time in implementing the sampling methods/routines, so wonder if its a good idea to remove it completely.

Reply:
It's explained in the message of the first commit.
When needed again, it will be really easy to recover thanks to git.